src/virtual-machine.js

```diff
@@ -460,6 +460,7 @@ class VirtualMachine extends EventEmitter {
             this.emitTargetsUpdate();
             this.emitWorkspaceUpdate();
             this.runtime.setEditingTarget(this.editingTarget);
+            this.runtime.emitTargetsInstalled(targets);
```

Comment:
Perhaps a note for the future... `installTargets` has the potential to get called multiple times for a single project.

`installTargets` gets called when a project gets loaded, but also whenever a sprite gets added or a deleted sprite gets restored. Is that okay?

I could see this maybe creating an issue down the line where we set extension state when all the targets are installed, and then accidentally reset it when adding a sprite... 
I'm not sure if it will actually be an issue, but just wanted to make sure you were aware of all the use cases of `installTargets`.

Reply:
Good call, I noticed that `installTargets` was being called each time a sprite was added as you mentioned, and I'm also not sure whether this is desirable not. I could imagine a use case where someone would want to know whenever a target is installed, though I can imagine more where it gets unintentionally called. 

In thinking about it again, is there a VM event that runs just once when a project has finished its initial load? Perhaps we can propagate that to the runtime? I've taken a look at `virtual-machine.js` to find an event that might work, and the closest seems to be `workspaceUpdate`, though that also runs whenever `installTargets` ones.

Reply:
> In thinking about it again, is there a VM event that runs just once when a project has finished its initial load? Perhaps we can propagate that to the runtime? I've taken a look at `virtual-machine.js` to find an event that might work, and the closest seems to be `workspaceUpdate`, though that also runs whenever `installTargets` ones.

There isn't already such an event that exists, but you should be able to achieve what you described by moving the event emitter line from the end of `installTargets` to the end of `loadProject` instead. You can chain another `.then` clause after this one that waits for the `deserializeProject` call to finish:
`.then(validatedInput => this.deserializeProject(validatedInput[0], validatedInput[1]))`